bentoml/_internal/server/instruments.py

```diff
@@ -41,23 +41,63 @@ def _setup(
         ],
     ):
         self.metrics_client = metrics_client
-        service_name = self.bento_service.name
+
+        DEFAULT_NAMESPACE = "bentoml_api_server"
+        if self.namespace == DEFAULT_NAMESPACE:
+            legacy_namespace = "BENTOML"
+        else:
+            legacy_namespace = self.namespace
+
+        # legacy metrics names for bentoml<1.0.6
+        self.legacy_metrics_request_duration = metrics_client.Histogram(
+            namespace=legacy_namespace,
+            name=metric_name(component_context.bento_name, "request_duration_seconds"),
+            documentation="Legacy API HTTP request duration in seconds",
+            labelnames=["endpoint", "service_version", "http_response_code"],
+            buckets=duration_buckets,
+        )
+        self.legacy_metrics_request_total = metrics_client.Counter(
+            namespace=legacy_namespace,
+            name=metric_name(component_context.bento_name, "request_total"),
+            documentation="Legacy total number of HTTP requests",
+            labelnames=["endpoint", "service_version", "http_response_code"],
+        )
+        self.legacy_metrics_request_in_progress = metrics_client.Gauge(
+            namespace=legacy_namespace,
+            name=metric_name(component_context.bento_name, "request_in_progress"),
+            documentation="Legacy total number of HTTP requests in progress now",
+            labelnames=["endpoint", "service_version"],
+            multiprocess_mode="livesum",
+        )
 
         self.metrics_request_duration = metrics_client.Histogram(
-            name=metric_name(service_name, "request_duration_seconds"),
+            namespace=self.namespace,
```

Comment:
Wouldn't allowing configuration of namespace make dashboard more difficult to create? @yetone

Reply:
I won't be a problem since we will make everything configurable through envvars